cirq/sim/clifford/clifford_simulator_test.py

```diff
@@ -291,6 +291,19 @@ def test_clifford_stabilizerStateChForm_repr():
     assert repr(state) == 'StabilizerStateChForm(num_qubits=2)'
 
 
+def test_clifforf_circuit_SHSYSHS():
+    q0 = cirq.LineQubit(0)
+    circuit = cirq.Circuit(cirq.S(q0), cirq.H(q0), cirq.S(q0), cirq.Y(q0),
+                           cirq.S(q0), cirq.H(q0), cirq.S(q0), cirq.measure(q0))
+
+    clifford_simulator = cirq.CliffordSimulator()
+    wave_function_simulator = cirq.Simulator()
+
+    np.testing.assert_almost_equal(
+        clifford_simulator.simulate(circuit).final_state.wave_function(),
```

Comment:
@viathor  Not related to this PR: What do you think about the fact that the clifford simulator needs a `.wave_function()` appended while the `wave_function_simulator` does not from a usability perpective? Should we make these consistent?

Reply:
Such inconsistency is definitely a problem in the API. Would you like to file an issue about this?

Thanks for catching.

Reply:
Why should there be an exponential cost in this otherwise polynomial simulator?
Generating the state vector is exponential in the number of qubits, but everything else is quadratic in the number of qubits? You would need to work around this.

Reply:
I agree exponential cost computation should not be executed implicitly in a surprising context. This does not conflict with API consistency. For example, some simulators already require that a user calls `wave_function` *explicitly* if they wish to see the reconstructed wavefunction.

In any case, none of this blocks this PR.